File: backend/src/main/java/com/skewwhiffy/auraltester/fraction/Factors.java
package com.skewwhiffy.auraltester.fraction;

public class Factors {
    public static int lcm(int first, int second) {
        if (first < 0) {
            return lcm(-first, second);
        }
        if (second < 0) {
            return lcm(first, -second);
        }
        return first * second / hcf(first, second);
    }

    public static int hcf(int first, int second) {
        if (first < 0) {
            return hcf(-first, second);
        }
        if (second < 0) {
            return hcf(first, -second);
        }
        if (first == 0) {
            return second;
        }
        if (second == 0) {
            return first;
        }
        if (first == second) {
            return first;
        }
        if (first > second) {
            return hcf(first - second, second);
        }
        return hcf(first, second - first);
    }

}

File: backend.old/src/main/kotlin/com/skewwhiffy/auraltester/fraction/Factors.kt
package com.skewwhiffy.auraltester.fraction

class Factors {
    companion object {
        fun lcm(first: Int, second: Int): Int {
            if (first < 0) {
                return lcm(-first, second)
            }
            return if (second < 0) {
                lcm(first, -second)
            } else first * second / hcf(first, second)
        }

        fun hcf(first: Int, second: Int): Int {
            if (first < 0) {
                return hcf(-first, second)
            }
            if (second < 0) {
                return hcf(first, -second)
            }
            if (first == 0) {
                return second
            }
            if (second == 0) {
                return first
            }
            if (first == second) {
                return first
            }
            return if (first > second) {
                hcf(first - second, second)
            } else hcf(first, second - first)
        }
    }
}